### PR TITLE
In gwas_linear_regression allow empty covariates #509

### LINKED ISSUE #509: Allow empty covariate list in gwas_linear_regression
In #463, the linear regression is first run with no covariates (i.e. only with an intercept). This is not possible with the current implementation of `gwas_linear_regression`, so we should add that capability. (There should be a check to see that `add_intercept` is True in the case of no covariates.)

### PR DESCRIPTION
**Pull request by tomwhite**
Fixes #509

### CODE REVIEW

**Comment by jeromekelleher** [comment]
One a side note here @tomwhite, I was looking to see if we had full test coverage here and I'm not seeing a codecov report. Is it just this PR or is this broken generally?

**Comment by tomwhite** [comment]
@jeromekelleher coverage reports are being run and uploaded (https://app.codecov.io/gh/pystatgen/sgkit/pulls?page=1&state=open&order=-pullid) and coverage is 100%. However, there seem to be some problems reported as "Missing base report" and "Unable to find commit in Github", which is perhaps why we don't get reports posted on all PRs here. Opened #514

File: sgkit/stats/association.py [hunk 1]
```diff
@@ -209,9 +209,15 @@ def gwas_linear_regression(
 
     G = _get_loop_covariates(ds, dosage=dosage, call_genotype=call_genotype)
 
-    X = da.asarray(concat_2d(ds[list(covariates)], dims=("samples", "covariates")))
-    if add_intercept:
-        X = da.concatenate([da.ones((X.shape[0], 1), dtype=X.dtype), X], axis=1)
+    if len(covariates) == 0:
+        if add_intercept:
+            X = da.ones((ds.dims["samples"], 1), dtype=float)
```
Comment (eric-czech) [inline sgkit/stats/association.py:214]: Any objection to float32 here to avoid promotion if `G` is 32-bit as well?

Comment (tomwhite) [reply sgkit/stats/association.py:214]: Good call

**Pull request merged**